pennylane/math/interface_utils.py

```diff
@@ -197,22 +209,24 @@ def get_deep_interface(value):
     return _get_interface_of_single_tensor(itr)
 
 
-def get_canonical_interface_name(interface):
+def get_canonical_interface_name(user_input: Union[str, Interface]) -> Interface:
     """Helper function to get the canonical interface name.
 
     Args:
-        interface (str): original interface to use as reference
+        interface (str, Interface): reference interface
 
     Raises:
         ValueError: key does not exist in the interface map
 
     Returns:
-        str: new interface
+        Interface: canonical interface
     """
 
     try:
-        return INTERFACE_MAP[interface]
+        if user_input in SUPPORTED_INTERFACE_NAMES:
+            return user_input
```

Comment:
super minor, but could we move these two lines up, so that only the code that might raise the error is inside the try-except clause?

Reply:
Good catch! 🙂 I'll do that.

Reply:
Address in latest commit, resolving!